src/components/DragAndDrop/Sortable.tsx

```diff
@@ -0,0 +1,127 @@
+
+  if (!maxNoteCollision) return false;
+  const collisionRatio = maxNoteCollision.data?.value;
+
+  return maxNoteCollision?.id === newId && !isActive(id, active) && collisionRatio && collisionRatio >= COMBINE_THRESHOLD && collisionRatio < MOVE_THRESHOLD;
+};
+
+export const shouldStack = (id: UniqueIdentifier, items: UniqueIdentifier[], newIndex: number, collisions: Collision[] | null, active: Active | null) => {
+  if (id !== active?.id) return false;
+  // if (items?.[newIndex] !== active.id) return false;
+  const noteCollisions = collisions?.filter((collision) => !isColumn(collision));
+  const maxNoteCollision = noteCollisions?.at(noteCollisions.length > 1 ? 1 : 0);
+
+  if (!maxNoteCollision || maxNoteCollision.id === active?.id) return false;
+
+  const collisionRatio = maxNoteCollision.data?.value;
+
+  return collisionRatio && collisionRatio >= COMBINE_THRESHOLD && collisionRatio < MOVE_THRESHOLD;
+};
+
+export const Sortable = ({id, children, disabled, className, columnId, setItems}: SortableProps) => {
+  const {collisions} = useDndContext();
+  const {setNodeRef, attributes, listeners, transition, transform, isDragging, items, newIndex, active} = useSortable({
+    id,
+    data: {columnId, type: "note"},
+    disabled,
+  });
+
+  const combine = shouldCombine(id, items, newIndex, collisions, active);
+
+  const globalNotes = useAppSelector((state) => state.notes);
+
+  const topCollision = collisions?.at(0);
+  /* const hasActive = !!active && items.includes(active.id.toString());
+  const hasOver = !!topCollision && items.includes(topCollision.id.toString()); */
+
+  const [localItems, setLocalItems] = useState(items);
+
+  useDndMonitor({
+    onDragStart: () => {
+      setLocalItems(items);
+    },
+    onDragEnd: (event: DragEndEvent) => {
+      if (!columnId || !active) return;
+
+      if (shouldStack(id, items, newIndex, collisions, active)) {
+        // combine notes
+        const noteCollisions = collisions?.filter((collision) => !isColumn(collision));
+        const maxNoteCollision = noteCollisions?.at(noteCollisions.length > 1 ? 1 : 0);
+        if (!maxNoteCollision) return;
+
+        const note = globalNotes.find((n) => n.id === maxNoteCollision.id.toString());
+        const position = {
+          stack: maxNoteCollision.id.toString(),
+          column: note?.position.column ?? columnId,
+          rank: note?.position.rank ?? 0,
+        };
+
+        store.dispatch(Actions.editNote(active.id.toString(), {position}));
+        if (setItems) setItems(items.filter((item) => item !== active.id.toString()).map((item) => item.toString()));
+
+        return;
+      }
+
+      if (!isActive(id, active)) return;
+
+      if (collisions && topCollision) {
+        if (items !== localItems || isColumn(topCollision)) {
+          const position = {
+            column: event.collisions?.find((collision) => isColumn(collision))?.id.toString() ?? columnId,
```

Comment:
Should be the same as just passing the function:
```suggestion
            column: event.collisions?.find(isColumn)?.id.toString() ?? columnId,
```